runtime/compilers/rillv1/parse_dotenv.go

```diff
@@ -2,25 +2,50 @@ package rillv1
 
 import (
 	"context"
+	"fmt"
+	"maps"
 	"os"
 
 	"github.com/joho/godotenv"
 )
 
-// parseDotEnv parses the env file present at repo root
-func (p *Parser) parseDotEnv(ctx context.Context, path string) error {
+// parseAndMergeEnvironmentVariables parses the .env file at the given path and merges it with the existing env vars
+func (p *Parser) parseAndMergeEnvironmentVariables(ctx context.Context, path string) error {
+	// Unlikely, but if the path is empty, return an error
+	if path == "" {
+		return fmt.Errorf("empty path provided for .env file")
+	}
+
+	// Read the .env file
 	data, err := p.Repo.Get(ctx, path)
 	if err != nil {
+		// If the file doesn't exist, return nil to exit early
 		if os.IsNotExist(err) {
 			return nil
 		}
 		return err
 	}
+
+	// Unmarshal the .env file
 	envMap, err := godotenv.Unmarshal(data)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to unmarshal .env file at %s: %w", path, err)
 	}
 
-	p.DotEnv = envMap
+	// Check if the map exists in the current state
+	currentEnv, ok := p.DotEnv[path]
+	if !ok {
+		p.DotEnv[path] = envMap
+		return nil
+	}
+
+	// Check if the map is the same as the current state
+	if maps.Equal(currentEnv, envMap) {
+		return nil
+	}
+
+	// Overwrite the current state with the new state
+	p.DotEnv[path] = envMap
```

Comment:
Isn't this the same as just doing `p.DotEnv[path] = envMap` outright?

Reply:
I see what you mean, is it worth just reassigning always?

Reply:
Updating in the map here is not expensive at this point (the assignment is very cheap, the garbage collection of one of the maps unavoidable at this point). Doing a `maps.Equal` will just be extra cycles.

The place where we don't want unnecessary updates is upstream, i.e. we don't want to the `Diff` to indicate the `.env` was changed if it wasn't, since that causes some very expensive upstream operations (reconciliation of all resources since we don't know which resources use a given env key or not).